abdiff/core/download_input_files.py

```diff
@@ -0,0 +1,85 @@
+import logging
+import subprocess
+
+import boto3
+from botocore.exceptions import ClientError
+from mypy_boto3_s3.client import S3Client
+
+from abdiff.config import Config
+
+logger = logging.getLogger(__name__)
+
+CONFIG = Config()
+
+
+def download_input_files(input_files: list[str]) -> None:
```

Comment:
Commenting generally on this function.

It's possible / likely / happened, that I suggested to @jonavellecuerdo that we keep this simple for now and perform these downloads sequentially.  That said, observing that this would be ripe for threaded (concurrent) downloads.  It would speed this up _considerably_, particularly if we are downloading 1.6k+ files @ 50+ gb.

@ehanson8 , @jonavellecuerdo - would you prefer to create a ticket for add concurrency here?  or have that implemented now?

Reply:
Feels like a ticket to me unless very simple to implement

Reply:
Created ticket [TIMX-386](https://mitlibraries.atlassian.net/browse/TIMX-386). For additional context for this decision, see https://github.com/MITLibraries/transmogrifier-ab-diff/pull/49#discussion_r1833038545.